src/Microsoft.DotNet.Build.Tasks/PackageFiles/ReferenceAssemblies.targets

```diff
@@ -37,4 +35,25 @@
     <AssemblyInfoLines Include="[assembly:System.Runtime.CompilerServices.ReferenceAssembly]" />
     <AssemblyInfoLines Include="[assembly:System.Reflection.AssemblyFlags((System.Reflection.AssemblyNameFlags)0x70)]" />
   </ItemGroup>
+
+  <!-- Roslyn requires the runtimemetadataversion parameter when building a core assembly -->
+  <Target Name="AddCoreAssemblyResponseFile" Condition="'$(IsCoreAssembly)' == 'true' AND '$(_usingRoslyn)' == 'true'" BeforeTargets="CoreCompile">
+    <CallTarget Targets="MergeCoreAssemblyResponseFile" Condition="'$(CompilerResponseFile)'!=''" />
+    <PropertyGroup>
+      <CompilerResponseFile Condition="'$(CompilerResponseFile)'==''">$(MSBuildThisFileDirectory)coreAssembly.rsp</CompilerResponseFile>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="MergeCoreAssemblyResponseFile">
+    <ItemGroup>
+      <ResponseFilesToMerge Include="$(CompilerResponseFile);$(MSBuildThisFileDirectory)coreAssembly.rsp" />
+    </ItemGroup>
+    <ReadLinesFromFile File="%(ResponseFilesToMerge.Identity)">
+      <Output TaskParameter="Lines" ItemName="ResponseFileLines" />
+    </ReadLinesFromFile>
+    <WriteLinesToFile File="$(IntermediateOutputPath)/resp.rsp" Lines="@(ResponseFileLines)" Overwrite="true" />
+    <PropertyGroup>
+      <CompilerResponseFile>$(IntermediateOutputPath)/resp.rsp</CompilerResponseFile>
```

Comment:
You are right. Looks like this works when you use a semicolon as a separator, I believe that in my testing I used a colon by mistake which led me to think it didn't work.